Tutorial: Get a user by email (#57)
UWC2-APIDOC/to-do-service-public: docs/tutorials/get-user-by-email.md

This tutorial assumes you have set the `base_url` variable in Postman to `http://localhost:3000`. If you are not using variables in Postman, replace all occurrences of \{\{base_url\}\} in these examples with `http://localhost:3000`. For more information on using environments and variables in Postman, see [Using Variables inside Postman](https://blog.postman.com/using-variables-inside-postman-and-collection-runner/).

## Get a user by email

You can get a user by their email address by making a `GET` request to the user resource in the To-Do service.

Follow these steps to get a user by email:

1. If your local service is not running, start it:

   ```shell
   cd <your-github-workspace>/to-do-service/api
   # Run the service and monitor its database file for updates
   json-server -w to-do-db-source.json
   ```
    
   * Replace `<your-github-workspace>` with the path to your To-Do service clone on your system.

2. Open the Postman app on your desktop.
3. Click the **New** button in Postman and select **HTTP** to create a new request.
4. Ensure **GET** is selected from the dropdown menu next to the request URL.

> 5. To search by a specific email, in the URL field, (1) enter the base URL for the To-Do service, (2) followed by the endpoint `/users`. Next, (3) type a question mark (`?`), followed by (4) the `email` query parameter, (5) the equals sign (`=`), and (6) the desired email address.

Reviewer: ```suggestion
5. To search by a specific email, in the URL field:
    1. Enter the base URL for the To-Do service, followed by the endpoint `/users`. 
    2. After the endpoint, type a question mark (`?`), followed by the `email` query parameter, the equals sign (`=`), and the desired email address.
    
The following is an example of such a URL:

```

It's a little easier to read when broken into a list.
The example is a good idea!

# Tutorial: Get a user by email

> In this tutorial, you will learn how to get a user by their email address from the To-Do service.

Reviewer: ```suggestion
In this tutorial, you will learn how to get a user from the To-Do service by their email address.
```

Author: Included in my newer commmit: "Revised based on feedback from rbwatson"

## Get a user by email

You can get a user by their email address by making a `GET` request to the user resource in the To-Do service.

Follow these steps to get a user by email:

1. If your local service is not running, start it:

   ```shell
   cd <your-github-workspace>/to-do-service/api
   # Run the service and monitor its database file for updates
   json-server -w to-do-db-source.json
   ```
    
   * Replace `<your-github-workspace>` with the path to your To-Do service clone on your system.

2. Open the Postman app on your desktop.
3. Click the **New** button in Postman and select **HTTP** to create a new request.
4. Ensure **GET** is selected from the dropdown menu next to the request URL.
5. In the URL field, enter the base URL for the To-Do service followed by the endpoint `/users`, for example:
<!-- {% raw %} -->
```

>    {{base_url}}/users

Reviewer: ```suggestion
   {{base_url}}/users?email=<email address>
```
re: the next comment

Author: Included in my newer commmit: "Revised based on feedback from rbwatson"

```
<!-- {% endraw %}) -->

> 6. To search for a specific user, append a question mark (`?`) to `/users` in the URL, followed by the relevant query parameter, and then click **Send** to submit the `GET` request to the To-Do service.

Reviewer: Could you demonstrate this in the URL example above?
That would seem to be easier to visualize than the textual description of the query parameter.

Author: Included in my newer commmit: "Revised based on feedback from rbwatson"

>    * To search by a specific email, add the `email` parameter to the URL followed by an equals sign (`=`) and the desired email address.

Reviewer: This, too, seems like it could be eliminated by updating the example above, since this tutorial is only about finding by email.

Author: Included in my newer commmit: "Revised based on feedback from rbwatson"

     **Example Request**
<!-- {% raw %} -->
```
     {{base_url}}/users?email=j.smith@example.com
```
<!-- {% endraw %}) -->
     **Example Response**

     Watch the response body, which should look something like this.

     ```json
     [
         {
              "last_name": "Smith",
              "first_name": "John",
              "email": "j.smith@example.com",
              "id": "1"
         }
     ]
     ```

     **Note:** If no user matches the search criteria, the service will return an empty array (`[]`), indicating that there are no users with the specified email address.

>      This completes the tutorial of getting a user by their email address. 

Reviewer: Great conclusion!
